### PR TITLE
Cleanup: Un-ignore some files from eslint

### PR DESCRIPTION
**Pull request by IanVS**
Issue:

As I was poking around I found a few files being ignored from eslint that don't need to be.  Un-ignoring them turned up a few small bugs as well.

## What I did

Removed `.d.ts` and `lib/cli/test` files from being ignored by ESLint, and fixed the linting errors that were exposed.

## How to test

CI

### CODE REVIEW

File: code/lib/source-loader/extract-source.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
 // DEPRECATED
 // Do not import @storybook/source-loader/extract-source directly.
 // Import @storybook/source-loader instead.
-export * from './dist/types/extract-source.d';
\ No newline at end of file
+export * from './dist/index';
```
Comment (IanVS) [inline code/lib/source-loader/extract-source.d.ts:4]: This likely broke in the move to use tsup to bundle source-loader.  It's deprecated though, so not a big deal I think.

File: code/addons/links/react.d.ts [hunk 2]
```diff
@@ -1,2 +1,2 @@
-export * from './dist/types/react';
-export { default } from './dist/types/react';
+export * from './dist/react';
+export { default } from './dist/react';
```
Comment (IanVS) [inline code/addons/links/react.d.ts:2]: I think this changed in the move to tsup

File: code/lib/source-loader/extract-source.d.ts (hunk 1 above)
Comment (kasperpeulen) [reply code/lib/source-loader/extract-source.d.ts:4]: Why index, instead of extract-source.d?

**Comment by kasperpeulen** [comment]
LGTM!

**Review by kasperpeulen — APPROVED** [review]

**Pull request merged**